### PR TITLE
fix: add wildcard for custody keyring

### PR DESCRIPTION
**Pull request by montelaidev**
## Explanation

This pr resolves the special case when passing a custody keyring type to the function keyringTypeToName. 

## References

Fixes: https://github.com/MetaMask/accounts-planning/issues/233

### `@metamask/keyring-controller`

- **CHANGED**:  Custody keyring type. 

## Checklist

- [x] I've updated the test suite for new or updated code as appropriate
- [x] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [ ] I've highlighted breaking changes using the "BREAKING" category above as appropriate

### CODE REVIEW

File: packages/accounts-controller/src/utils.ts [hunk 1]
```diff
@@ -9,6 +9,12 @@ import { v4 as uuid } from 'uuid';
  * @returns The name of the keyring type.
  */
 export function keyringTypeToName(keyringType: string): string {
+  // Custody keyrings are a special case, as they are not a single type
+  // they just start with the prefix `Custody`
+  if (/Custody.*/u.test(keyringType)) {
```
[the remark this answers is not in the document]
Comment (montelaidev) [reply packages/accounts-controller/src/utils.ts]: Thanks, the startsWith is a better check.

**Review by gantunesr — APPROVED** [review]
lgtm

**Comment by gantunesr** [comment]
Do you think instead of this method keyrings should have a default name and provide a get method to expose it? That way we wouldn't need to worry about this mapping and we can make it part of the API

**Comment by montelaidev** [comment]
> Do you think instead of this method keyrings should have a default account name and provide a get method to expose it? That way we wouldn't need to worry about this mapping and we can make it part of the API

Its a good idea, lets plan this out when we refactor the keyrings to use the keyring api

**Pull request merged**